packages/codemod/transformations/core/v2-to-v3/featured-icon-migration.ts

```diff
@@ -0,0 +1,29 @@
+import { wrap, getComponentNameOrAliasFromImports, getImports } from "../../../src/utils";
+import { TransformationContext } from "../../../types";
+import { CORE_ICONS_IMPORT_PATH } from "../../../src/consts";
+
+/**
+ * 1. Replace Featured icon with Upgrade icon
+ */
+function transform({ j, root }: TransformationContext) {
+  const imports = getImports(root, CORE_ICONS_IMPORT_PATH);
+
+  if (!imports.length) return;
+
+  const componentName = getComponentNameOrAliasFromImports(j, imports, "Featured");
+  if (!componentName) return;
+
+  imports.forEach(path => {
+    path.node.specifiers?.forEach(spec => {
+      if (j.ImportSpecifier.check(spec)) {
+        if (spec.imported.name === "Featured") {
+          spec.imported.name = "Upgrade";
+        }
+      }
+    });
+  });
+
+  root.find(j.JSXExpressionContainer).find(j.Identifier, { name: componentName }).replaceWith(j.identifier("Upgrade"));
```

Comment:
The following are stuff I **think** would happen, but worth adding as tests to make sure they do not occur:

this would replace also
```
<SomeComponent myProp={"Featured"} />
```
to this
```
<SomeComponent myProp={"Upgrade"} />
```

and also this
```
<SomeComponent myProp={Featured} />
```
to this
```
<SomeComponent myProp={Upgrade} />
```

basically everything that is inside a prop which is inside `{}` and includes the name Featured.
e.g.
```
<AnotherComponent thisIsAnObject={{ oneAttr: "someValue", Featured: true }} />
```

can you add tests with those cases and check it?